### PR TITLE
TN-1953-IRONMAN-widen-QB-windows-wip

### PR DESCRIPTION
**Pull request by mcjustin**
https://jira.movember.com/browse/TN-1953 @pbugni  this is just adjusting the baseline and 3 mo, to get an idea of the change. Will adjust their communication requests next.

### CODE REVIEW

**Comment by mcjustin** [comment]
@pbugni  just noting that PCCTC wants to change things a bit more than these initial commits would indicate (eg they want to change baseline too, which of course ripples into the 3 mo). But, can you do a brief review of the existing commits to see if my approach here raises any flags?

**Comment by mcjustin** [comment]
Per travis, I seem to have syntax issues, will address later today...

File: portal/config/eproms/QuestionnaireBank.json [hunk 1]
```diff
@@ -180,10 +180,10 @@
     },
     {
       "classification": "recurring",
-      "due": "{\"days\": 14}",
-      "expired": "{\"days\": 29}",
+      "due": "{\"months\": 1, \"days\": 14}",
```
Comment (pbugni) [inline portal/config/eproms/QuestionnaireBank.json]: I don't believe this is what you want.  The docs for this are hidden in the source code, I see.  Most everything is *relative* to `start`.  In fact, I think you may want due to be days=0.

File: portal/config/eproms/QuestionnaireBank.json [hunk 2]
```diff
@@ -216,10 +216,10 @@
...
```
Comment (pbugni) [inline portal/config/eproms/QuestionnaireBank.json]: Same comment as above.  I think we should try `"due": "{\"days\": 0}"`

File: portal/config/eproms/CommunicationRequest.json [hunk 3]
```diff
@@ -377,7 +377,7 @@
         }
       ],
       "lr_uuid": "b71dfdf6-bb19-3fff-84bd-4cdb30050499",
-      "notify_post_qb_start": "{\"days\": 0}",
+      "notify_post_qb_start": "{\"month\": 1, \"days\": 0}",
```
Comment (pbugni) [inline portal/config/eproms/CommunicationRequest.json]: The internals of RelativeDelta are a pain - to get around month versus months comparison issues, we enforce all such nouns to be in their plural form - that's why the tests are failing

File: portal/config/eproms/QuestionnaireBank.json (hunk 1 above)
Comment (mcjustin) [reply portal/config/eproms/QuestionnaireBank.json]: Thanks. Hmm, entry.start is currently set to 0 for all... Do you mean relative to entry.recurs.start ? I'll take another swing at this once the windows requirements have settled (hopefully tomorrow)

Comment (pbugni) [reply portal/config/eproms/QuestionnaireBank.json]: Correct.  Here's the significant code, to determine the "start" of a "recurring QB" (`self`), where `ic` == iteration count, and `trigger_date` is the initial consent:
```
                start = (trigger_date + RelativeDelta(self.start) +
                         RelativeDelta(recur.start) +
                         (ic * RelativeDelta(recur.cycle_length)))
```

**Comment by mcjustin** [comment]
Scuttling this in favor of https://github.com/uwcirg/truenth-portal/pull/3108  (prefer to branch of master)

**Comment by pbugni** [comment]
Closing in favor of #3108

**Pull request closed without merging**